components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/cdc/CDCProcessor.java

```diff
@@ -0,0 +1,297 @@
+            file.createNewFile();
+        }
+    }
+
+    private static synchronized String resolveSecureVault(MessageContext messageContext, String passwordString) {
+        if (passwordString == null) {
+            return null;
+        }
+        Matcher lookupMatcher = vaultLookupPattern.matcher(passwordString);
+        String resolvedValue = "";
+        if (lookupMatcher.find()) {
+            Value expression;
+            String expressionStr = lookupMatcher.group(1);
+            try {
+                expression = new Value(new SynapseXPath(expressionStr));
+
+            } catch (JaxenException e) {
+                throw new SynapseException("Error while building the expression : " + expressionStr, e);
+            }
+            resolvedValue = expression.evaluateValue(messageContext);
+            if (StringUtils.isEmpty(resolvedValue)) {
+                LOGGER.warn("Found Empty value for expression : " + expression.getExpression());
+                resolvedValue = "";
+            }
+        } else {
+            resolvedValue = passwordString;
+        }
+        return resolvedValue;
+    }
+
+
+    /**
+     * This will be called at the time of synapse artifact deployment.
+     */
+    public void init() {
+        LOGGER.info("Inbound CDC listener " + name + " starting ...");
+        pollingConsumer = new CDCPollingConsumer(cdcProperties, name, synapseEnvironment, interval);
+        pollingConsumer.registerHandler(new CDCInjectHandler(injectingSeq, onErrorSeq, sequential,
+                synapseEnvironment, cdcProperties));
+        start();
+    }
+
+
+    /**
+     * Register/start the schedule service
+     */
+    public void start() {
+        InboundTask task = new CDCTask(pollingConsumer, interval);
+        start(task, ENDPOINT_POSTFIX);
+    }
+
+    /**
+     * Remove inbound endpoints.
+     *
+     * @param removeTask Whether to remove scheduled task from the registry or not.
+     */
+    @Override
+    public void destroy(boolean removeTask) {
+        if (removeTask) {
+            destroy();
+            pollingConsumer.destroy();
+        }
+    }
+
+    @Override
+    public void update() {
+        // This will not be called for inbound endpoints
+    }
+
+    private String getOpCode(String op) {
+        if (op != null) {
+            switch (operations.valueOf(op)) {
+                case create:
+                    return "c";
+                case update:
+                    return "u";
+                case delete:
+                    return "d";
+                case truncate:
+                    return "t";
+            }
+        }
+        return "";
+    }
+
+    /**
+     * Get the comma separated list containing allowed operations and returns the string of allowed operation codes
+     * @param allowedOperationsString string
+     * @return the coma separated string of skipped operation codes
+     */
+    private String getSkippedOperationsString(String allowedOperationsString) {
```

Comment:
Shall we simplify this method a bit?
Something like
```java
private String getSkippedOperationsString(String allowedOperationsString) {
    Set<String> allOperations = Stream.of(operations.create.toString(), /* ... others... */).collect(Collectors.toSet());
    Set<String> allowedOperations = /* Split `allowedOperationsString`, trim, get the opCode and populate this */ ;
    allOperations.removeAll(allowedOperations);
    return String.join(",", allOperations);
}
```

Reply:
Fixed